### ISSUE TITLE
[Java 9+ support] Illegal reflective access warnings

### ISSUE THREAD

**Issue by eau-de-la-seine** [jruby/jruby#5913]
Hi, I have "illegal reflective access" warnings when I use JRuby with Java9+

### Environment

* JRuby version: **9.2.8.0**, imported with Maven
* Operating system: Windows 10
* JDK: tested with JDK11 (embedded in IntelliJ) and OpenJDK13

### Expected Behavior

No warning

### Actual Behavior

Shows the following warning:
```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.headius.backport9.modules.Modules (file:/C:/Users/gokan/.m2/repository/com/headius/backport9/1.3/backport9-1.3.jar) to field java.io.FileDescriptor.fd
WARNING: Please consider reporting this to the maintainers of com.headius.backport9.modules.Modules
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

**Comment by headius** [comment]
Could you try a snapshot build from jruby.org? Master should behave better than 9.2.8.

**Comment by headius** [comment]
I'm going to optimistically call this a duplicate of other issues fixed by better module handling in 9.2.9.
